RiotNSE/NotificationService.swift

```diff
@@ -232,21 +232,51 @@ class NotificationService: UNNotificationServiceExtension {
         // At this stage we don't know the message type, so leave the body as set in didReceive.
     }
     
-    private func fetchEvent(withEventId eventId: String, roomId: String, allowSync: Bool = true) {
-        MXLog.debug("[NotificationService] fetchEvent")
-        
-        NotificationService.backgroundSyncService.event(withEventId: eventId,
-                                                        inRoom: roomId,
-                                                        completion: { (response) in
-                                                            switch response {
-                                                            case .success(let event):
-                                                                MXLog.debug("[NotificationService] fetchEvent: Event fetched successfully")
-                                                                self.processEvent(event)
-                                                            case .failure(let error):
-                                                                MXLog.debug("[NotificationService] fetchEvent: error: \(error)")
-                                                                self.fallbackToBestAttemptContent(forEventId: eventId)
-                                                            }
-                                                        })
+    private func fetchAndProcessEvent(withEventId eventId: String, roomId: String) {
+        MXLog.debug("[NotificationService] fetchAndProcessEvent")
+                
+        NotificationService.backgroundSyncService.event(withEventId: eventId, inRoom: roomId) { [weak self] (response) in
+            switch response {
+            case .success(let event):
+                MXLog.debug("[NotificationService] fetchAndProcessEvent: Event fetched successfully")
+                self?.checkPlaybackAndContinueProcessing(event, roomId: roomId)
+            case .failure(let error):
+                MXLog.error("[NotificationService] fetchAndProcessEvent: Failed fetching notification event with error: \(error)")
+                self?.fallbackToBestAttemptContent(forEventId: eventId)
+            }
+        }
+    }
+    
+    private func checkPlaybackAndContinueProcessing(_ notificationEvent: MXEvent, roomId: String) {
+        NotificationService.backgroundSyncService.roomAccountData(forRoomId: roomId) { [weak self] response in
+            switch response {
+            case .success(let roomAccountData):
+                NotificationService.backgroundSyncService.event(withEventId: roomAccountData.readMarkerEventId, inRoom: roomId) { (response) in
```

Comment:
`MXBackgroundSyncService.event` api was not designed to fetch _any_ event actually. We may end up with unexpected `/sync` requests with this. I'd suggest to create another api for this in `MXBackgroundSyncService`, by probably just allowing `/event` requests in it.

Reply:
Unfortunately I don't think we'll see the Sygnal part come in very soon. I closed my original PR as it was incomplete and now the backend team needs to pick it up and do it properly. I'll investigate the /sync /events thing though.